test/Microsoft.Azure.Relay.UnitTests/HybridRequestTests.cs

```diff
@@ -275,13 +274,83 @@ async Task LargeRequestEmptyResponse(EndpointTestType endpointTestType)
             }
         }
 
+        [Fact, DisplayTestMethodName]
+        async Task LoadBalancedListeners_WebRequest()
+        {
+            EndpointTestType endpointTestType = EndpointTestType.Authenticated;
+
+            // Allow more than two connections to a destination when using HttpWebRequests
+            ServicePointManager.DefaultConnectionLimit = 200;
+
+            Func<Uri, TokenProvider, IEnumerable<int>, CancellationToken, Task> sendBatchFunc = async (httpUri, tokenProvider, indexes, cancelToken) =>
+            {
+                try
+                {
+                    foreach (var index in indexes)
+                    {
+                        var httpRequest = (HttpWebRequest)WebRequest.Create(httpUri);
+                        using (var abortRegistration = cancelToken.Register(() => httpRequest.Abort()))
+                        {
+                            httpRequest.Method = HttpMethod.Get.Method;
+                            await AddAuthorizationHeader(tokenProvider, httpRequest.Headers, httpUri);
+
+                            using (var httpResponse = (HttpWebResponse)(await httpRequest.GetResponseAsync()))
+                            using (var responseStream = httpResponse.GetResponseStream())
+                            {
+                                Assert.Equal(HttpStatusCode.OK, httpResponse.StatusCode);
+                            }
+                        }
+                    }
+                }
+                catch (WebException webException)
+                {
+                    HttpStatusCode? status = (webException.Response as HttpWebResponse)?.StatusCode;
+                    string messageIndexes = indexes.First() + "-" + indexes.Last();
+                    TestUtility.Log($"Messages {messageIndexes} Error: {status} {webException}");
+                    throw;
+                }
+            };
+
+            await LoadBalancedListenersCore(endpointTestType, sendBatchFunc);
```

Comment:
Would this be less or more confusing if the lambda where written inline?
```
await LoadBalancedListenersCore(
    endpointTestType,
    async (httpUri, tokenProvider, indexes, cancelToken) =>
    {
        . . . 
     });
```

Reply:
I did this in iteration 3.

---
In reply to: [315829211](https://github.com/Azure/azure-relay-dotnet/pull/112#discussion_r315829211) [](ancestors = 315829211)